src/core/utilities/mpi_utils.cpp

```diff
@@ -53,5 +53,9 @@ std::string date_time(std::string format)
     return collect(date_time)[0];
 }
 
+std::int64_t unix_timestamp_now()
+{ // static cast as std::time_t typedef can vary across operating systems
+    return collect(static_cast<std::int64_t>(std::time(NULL)))[0];
```

Comment:
If I understand correctly, this gather the local unix time per process on all processes and then each process take the time of rank 0?
I wonder if that is what should be done. When using a very large number of MPI processes we will be transferring lots of data for just using 1 value

There is [MPI Broadcast](https://mpitutorial.com/tutorials/mpi-broadcast-and-collective-communication/) that can/should preferably used for that. It allows one rank, here rank 0, to send its value to all other processes.